iris-mpc-gpu/tests/e2e.rs

```diff
@@ -28,6 +28,20 @@ mod e2e_test {
     const MAX_DELETIONS_PER_BATCH: usize = 10;
     const THRESHOLD_ABSOLUTE: usize = 4800; // 0.375 * 12800
 
+    #[derive(Clone)]
+    pub struct E2ETemplate {
+        left:  IrisCode,
+        right: IrisCode,
+    }
+
+    #[derive(Clone)]
+    pub struct E2ESharedTemplate {
+        pub left_shared_code:  [GaloisRingIrisCodeShare; 3],
+        pub left_shared_mask:  Vec<GaloisRingTrimmedMaskCodeShare>,
```

Comment:
this could also be [GaloisRingTrimmedMaskCodeShare; 3] to unify and reduce confusion